java/src/test/java/io/cucumber/java/JavaDocStringTypeDefinitionTest.java

```diff
@@ -28,16 +96,39 @@
     private final DocStringTypeRegistryDocStringConverter converter = new DocStringTypeRegistryDocStringConverter(
         registry);
 
+    @Test
+    void doc_string_type_gets_correctly_init() throws NoSuchMethodException {
```

Comment:
This test is testing the implementation details of `DocString`. This requires making more `DocStrings` methods public then needed. Instead consider testing if the correct conversion method is invoked for simple and complex types (e.g. `List<Sting>` and `List`.